src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java

```diff
@@ -397,7 +397,7 @@ public StateFlow build(FlowState.Main initialState) {
                 .transitionTo(CLAIM_DETAILS_NOTIFIED_TIME_EXTENSION)
                     .onlyIf(respondentTimeExtension.and(not(notificationAcknowledged)))
                 //Acknowledging Claim First
-                .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledged)
+                .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledged.and(not(isInHearingReadiness)))
```

Comment:
Is this a separate error? Or related to the first?

Reply:
I assume it is to cover the DJ journey, or something?

Reply:
Unrelated, but found during the changes made to test.

          .takenOfflineDate(LocalDateTime.now())
                // ensure no ambiguous transitions between HEARING_READINESS and TAKEN_OFFLINE_AFTER_SDO
                .hearingReferenceNumber("12345")
                .listingOrRelisting(ListingOrRelisting.LISTING)
                .build();

If takenOfflineDate, was null, we would get a ambiguity between NOTIFICATION_ACKNOWLEDGED and HEARING_READINESS.

In practice not a possible scenario afaik, as CCD allowed states in events would not allow it

Reply:
Cool - makes sense :)